tcping.go

```diff
@@ -406,12 +459,22 @@ func tcping(tcpStats *stats) {
 			calculatedDowntime := calcTime(uint(math.Ceil(latestDowntimeDuration)))
 			color.Yellow.Printf("No response received for %s\n", calculatedDowntime)
 
-			tcpStats.endOfDowntime = getSystemTime()
-			calcLongestDowntime(tcpStats)
+			/* Update startOfUptime */
+			tcpStats.startOfUptime = getSystemTime()
+
+			/* Calculate the longest downtime */
+			endOfDowntime := getSystemTime()
+			calcLongestDowntime(tcpStats, endOfDowntime)
+			tcpStats.startOfDowntime = time.Time{}
 
 			tcpStats.wasDown = false
 		}
 
+		/* It means it is the first time to get a response*/
+		if tcpStats.startOfUptime.Format(timeFormat) == nullTimeFormat {
+			tcpStats.startOfUptime = getSystemTime()
+		}
```

Comment:
Can we not move this logic somewhere else? My intention is to keep this function more readable and have less logic in it.

Reply:
You are right. Maybe in the future we can work that out. For now, let's ship this PR.